Content.Client/CartridgeLoader/Cartridges/SpaceVendorsUiFragment.xaml.cs

```diff
@@ -0,0 +1,71 @@
+using Content.Shared.CartridgeLoader.Cartridges;
+using Robust.Client.AutoGenerated;
+using Robust.Client.Graphics;
+using Robust.Client.UserInterface.Controls;
+using Robust.Client.UserInterface.XAML;
+
+namespace Content.Client.CartridgeLoader.Cartridges;
+
+[GenerateTypedNameReferences]
+public sealed partial class SpaceVendorsUiFragment : BoxContainer
+{
+    private readonly StyleBoxFlat _styleBox = new()
+    {
+        BackgroundColor = Color.Transparent,
+        BorderColor = Color.FromHex("#5a5a5a"),
+        BorderThickness = new Thickness(0, 0, 0, 1)
+    };
+
+    public SpaceVendorsUiFragment()
+    {
+        RobustXamlLoader.Load(this);
+        Orientation = LayoutOrientation.Vertical;
+        HorizontalExpand = true;
+        VerticalExpand = true;
+        HeaderPanel.PanelOverride = _styleBox;
+    }
+
+    public void UpdateState(List<AppraisedItem> items)
+    {
+        SpaceVendorsContainer.RemoveAllChildren();
+
+        //Reverse the list so the oldest entries appear at the bottom
+        items.Reverse();
+
+        //Enable scrolling if there are more entries that can fit on the screen
+        ScrollContainer.HScrollEnabled = items.Count > 9;
+
+        foreach (var item in items)
+        {
+            AddProbedDevice(item);
+        }
+    }
+
+    private void AddProbedDevice(AppraisedItem item)
+    {
+        var row = new BoxContainer();
+        row.HorizontalExpand = true;
+        row.Orientation = LayoutOrientation.Horizontal;
+        row.Margin = new Thickness(3);
+
+        var nameLabel = new Label();
+        nameLabel.Text = item.Name;
+        nameLabel.HorizontalExpand = true;
+        nameLabel.ClipText = true;
+        row.AddChild(nameLabel);
+
+        var priceLabel = new Label();
+        priceLabel.Text = item.Price;
+        priceLabel.HorizontalExpand = true;
+        priceLabel.ClipText = true;
+        row.AddChild(priceLabel);
+
+        var minutesLabel = new Label();
+        minutesLabel.Text = item.Minutes.ToString()+" min";
+        minutesLabel.HorizontalExpand = true;
+        minutesLabel.ClipText = true;
+        row.AddChild(minutesLabel);
+
+        SpaceVendorsContainer.AddChild(row);
+    }
+}
```

Comment:
Use UpdateDraw to update the device timers every frame.

Also you'll want to subscribe to entityunpausedevent for the cartridge and update all of the timers with the added paused time and dirty it too.

Reply:
I did the update through the FrameUpdate method, I don't understand which UpdateDraw method you are talking about.

About entityunpausedevent, I did not understand what it is for and how to do it.